tests/python/openassetio/test_context.py

```diff
@@ -114,10 +113,11 @@ def test_when_set_to_None_then_returns_None(self, a_context):
 
 class Test_Context_managerState:
     def test_when_set_to_unknown_type_then_raises_TypeError(self, a_context):
-        with pytest.raises(ValueError) as err:
```

Comment:
[fix] Same comment re: test as above for assignment with a valid value.

Reply:
Added test in cfc8bb8a